k-distribution/src/main/scripts/lib/pyk/pyk/cterm.py

```diff
@@ -0,0 +1,105 @@
+from dataclasses import dataclass
+from functools import cached_property
+from itertools import chain
+from typing import Iterable, Optional, Set, Tuple
+
+from pyk.kast import (
+    KClaim,
+    KInner,
+    KRewrite,
+    KRule,
+    KRuleLike,
+    KVariable,
+    flattenLabel,
+)
+from pyk.kastManip import (
+    Subst,
+    count_rhs_vars,
+    count_vars,
+    drop_var_prefixes,
+    extract_lhs,
+    extract_rhs,
+    match,
+    pushDownRewrites,
+    simplifyBool,
+    splitConfigAndConstraints,
+    unsafeMlPredToBool,
+)
+from pyk.prelude import mlAnd
+
+
+@dataclass(frozen=True)
+class CTerm:
+    config: KInner  # TODO Optional?
+    constraints: Tuple[KInner, ...]
+
+    def __init__(self, term: KInner) -> None:
+        config, constraint = splitConfigAndConstraints(term)
+        constraints = tuple(flattenLabel('#And', constraint))
+        object.__setattr__(self, 'config', config)
+        object.__setattr__(self, 'constraints', constraints)
+
+    def __iter__(self):
+        return chain([self.config], self.constraints)
+
+    @cached_property
+    def term(self) -> KInner:
+        return mlAnd(self)
```

Comment:
`term` usually refers to something structural without constraints.

Maybe this should be `cterm(...)` or `pattern(...)`

Reply:
I wouldn't call it `pattern`, as that suggest pattern matching as context. I guess `cterm` is fine, though if `cterm: CTerm`, then `cterm.cterm` can be misleading, as `cterm.cterm: KInner`.